llama-index-integrations/vector_stores/llama-index-vector-stores-redis/llama_index/vector_stores/redis/base.py

```diff
@@ -230,76 +328,32 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
             redis.exceptions.TimeoutError: If there is a timeout querying the index.
             ValueError: If no documents are found when querying the index.
         """
-        return_fields = [
-            "id",
-            "doc_id",
-            "text",
-            self._vector_key,
-            "vector_score",
-            "_node_content",
-        ]
-
-        filters = _to_redis_filters(query.filters) if query.filters is not None else "*"
-
-        _logger.info(f"Using filters: {filters}")
-
-        redis_query = get_redis_query(
-            return_fields=return_fields,
-            top_k=query.similarity_top_k,
-            vector_field=self._vector_field,
-            filters=filters,
-        )
-
         if not query.query_embedding:
             raise ValueError("Query embedding is required for querying.")
 
-        query_params = {
-            "vector": array_to_buffer(query.query_embedding),
-        }
-        _logger.info(f"Querying index {self._index_name}")
+        redis_query = self._to_redis_query(query)
+        logger.info(
+            f"Querying index {self._index.name} with filters {redis_query.filters}"
+        )
 
         try:
-            results = self._redis_client.ft(self._index_name).search(
-                redis_query, query_params=query_params  # type: ignore
-            )
+            results = self._index.query(redis_query)
         except RedisTimeoutError as e:
-            _logger.error(f"Query timed out on {self._index_name}: {e}")
+            logger.error(f"Query timed out on {self._index.name}: {e}")
             raise
         except RedisError as e:
-            _logger.error(f"Error querying {self._index_name}: {e}")
+            logger.error(f"Error querying {self._index.name}: {e}")
             raise
 
-        if len(results.docs) == 0:
+        if not results:
             raise ValueError(
```

Comment:
tbh I'm kind of torn on raising an error here or not

If the user applies filters that for example return no documents, I feel like that should be ok? Thoughts?

Reply:
agreed!